powersms/powersms_core.py

```diff
@@ -15,6 +17,29 @@ def check_numbers(self, cr, uid, ids, numbers):
             return True
         return False
 
+    def _get_json_body(self, number_to, message, from_name, context=None):
+        special_characters = [
+            u"€",
+        ]
+        dict_sms = {
+            "txt": message,
+        }
+        if any(special_char in message for special_char in special_characters):
+            dict_sms = {
+                "charset":"utf-16",
+                "data_coding":"unicode",
+                "txt": base64.b64encode(message.encode('utf-16')),
+            }
+        json_body = {
+            "sms": dict_sms,
+            "dst": {
+                "num": number_to,
+            },
+            "src": from_name,
+        }
```

Comment:
Crec que està malament ja que abans quedava tot sota sms, seria:

```suggestion
        json_body = {"sms": dict(dst={"num": number_to}, src=from_name, **dict_sms)}
```